src/components/InputsTable.js

```diff
@@ -104,94 +107,97 @@ const InputsTable = props => {
-            size: { basis: 250, grow: 0 },
-            field: 'taskName',
-            headerRenderer: () => h(Sortable, { sort: inputTableSort, field: 'taskName', onSort: setInputTableSort }, [h(HeaderCell, ['Task name'])]),
-            cellRenderer: ({ rowIndex }) => {
-              return h(TextCell, { style: { fontWeight: 500 } }, [inputTableData[rowIndex].taskName])
-            }
-          },
-          {
-            size: { basis: 360, grow: 0 },
-            field: 'variable',
-            headerRenderer: () => h(Sortable, { sort: inputTableSort, field: 'variable', onSort: setInputTableSort }, [h(HeaderCell, ['Variable'])]),
-            cellRenderer: ({ rowIndex }) => {
-              return h(TextCell, { style: Utils.inputTypeStyle(inputTableData[rowIndex].input_type) }, [inputTableData[rowIndex].variable])
-            }
-          },
-          {
-            size: { basis: 160, grow: 0 },
-            field: 'inputTypeStr',
-            headerRenderer: () => h(HeaderCell, ['Type']),
-            cellRenderer: ({ rowIndex }) => {
-              return h(TextCell, { style: Utils.inputTypeStyle(inputTableData[rowIndex].input_type) }, [inputTableData[rowIndex].inputTypeStr])
-            }
-          },
-          {
-            size: { basis: 300, grow: 0 },
-            headerRenderer: () => h(HeaderCell, ['Input sources']),
-            cellRenderer: ({ rowIndex }) => {
-              return InputSourceSelect({
-                source: _.get('source', inputTableData[rowIndex]),
-                inputType: _.get('input_type', inputTableData[rowIndex]),
-                setSource: source => setConfiguredInputDefinition(
-                  _.set(`[${inputTableData[rowIndex].configurationIndex}].source`, source, configuredInputDefinition))
-              })
-            }
-          },
-          {
-            headerRenderer: () => h(HeaderCell, ['Attribute']),
-            cellRenderer: ({ rowIndex }) => {
-              const source = _.get(`${rowIndex}.source`, inputTableData)
-              const inputName = _.get(`${rowIndex}.input_name`, inputTableData)
-              return h(WithWarnings, {
-                baseComponent: Utils.switchCase(source.type || 'none',
-                  ['record_lookup', () => recordLookup(rowIndex)],
-                  ['literal', () => parameterValueSelect(rowIndex)],
-                  ['object_builder', () => structBuilderLink(rowIndex)],
-                  ['none', () => sourceNone(rowIndex)]
-                ),
-                message: _.find(message => message.name === inputName)(inputValidations)
-              })
-            }
+  return h(div, { style: { height: 'calc(100% - 3rem)' } }, [
+    h(InputsButtonRow, {
+      optionalButtonProps: {
+        includeOptionalInputs, setIncludeOptionalInputs
+      },
+      setFromDataTableButtonProps: {
+        inputRowsInDataTable, setConfiguredInputDefinition
+      },
+      searchProps: {
+        searchFilter, setSearchFilter
+      }
```

Comment:
It seems Terra UI has a component called `DelayedSearchInput` for filtering inputs. Maybe we can use that here so that we can reuse components especially since we are reintegrating back into Terra UI. One example of this component is in The `Inputs` tab in the Submission Config page in a GCP workspace:
<img width="1377" alt="Screenshot 2023-06-27 at 5 03 31 PM" src="https://github.com/DataBiosphere/cbas-ui/assets/16748522/656ba4af-e9fa-427b-b34b-a37fbd0c38a5">

Reply:
Yes, the component used here was copied over from the Terra-UI version so it should be pretty easy to merge back with Terra

Reply:
Ah I see. Apologies that I missed it.